### ISSUE TITLE
super[35251]: Error: Failed to send MDM install update/upgrade macOS request. Verify that this account has appropriate privileges

### ISSUE THREAD

**Issue by callumwebb10** [Macjutsu/super#172]
Two thirds of our machines have successfully been upgraded using v4 script with jamf mdm profiles/policy. Seemingly out of nowhere every remaining machine is now failing with the above error (regardless of current os, hardware version, etc.) 

I'm not suggesting it's an error with the script as it clearly worked fine, just wondering if anyone else has seen this issue..?

**Comment by Macjutsu** [comment]
That means the Jamf API failed to accept the update/upgrade command. More specifically, `super` asked the Jamf API to update/upgrade and then `super` did not receive an HTTP 200 or 201, which is the equivalent to "OK".

You'll have to acquire a super.log in `--verbose-mode` to see the specific error.

**Comment by callumwebb10** [comment]
Thanks for your swift response, super is trying to upgrade to 13.6.2 (latest Ventura release as of 7th Nov 23) as per my flag to target v13,  but doesnt seem to find the update to download. Please see below excerpts from logs

Super.log: --
Wed Nov 15 14:34:45 super[95718]: Verbose Mode: Function install_macos_mdm: install_macos_mdm_version is: 13.6.2

[...]

=== OS Update Item ===
Product Key:               042-95459
Title:                     macOS Ventura
Version:                   13.6.2 ((null))
Deferred:                  no  (Date: )
Tags:                      SUBUNDLE:com.apple.InstallAssistant.macOSVentura; CUSTOMER
Mac OS update:             no
Major OS Update:           YES
  Major Version:           13.6.2
  Major BundleID:          com.apple.InstallAssistant.macOSVentura
  Major Title:             macOS Ventura

[...]

   ASUListCache =     (
        "Safari17.1MontereyAuto",
        "MSU_UPDATE_21G920_patch_12.7.1",
        "MSU_UPDATE_23B81_patch_14.1.1"
    );


As you can see 13.6.2 is not being downloaded hence the install is failing, the only available updates are to safari17.1, MacOS12.7.1 and MacOS14.1.1
All machines prior to the 7th Nov completed successfully to 13.6.1 (we're trying to avoid Sonoma if possible). 
Is there an option to target a minor version? The logs don't explain why the software update tool can't get 13.6.2 but it can see that it exists - hence trying to upgrade to it.

**Comment by Macjutsu** [comment]
You've cut out a lot of the super.log... so while your conclusions may be correct about the output from `mdmclient`, that's not the only thing `super` does to find updates.

In fact, one of the fixes in `super` v4.0.1 is that it knows that macOS 13 is no longer available via `softwareupdate` and should be downloaded as a full installer via `mist-cli`.

**Comment by Macjutsu** [comment]
Also looking back further in this thread... your original issue was related to the API call failing... which is not directly related to what `super` does if it's going to pre-download an update or upgrade.

**Comment by domitreous** [comment]
So I noticed and this log confirms that 13.6.2 is being seen as a major update. If you have a defer major updates for X days it may not process. I think super itself should not have an issue but it would definately get blocked by that deferral.Sent from my iPhoneOn Nov 15, 2023, at 1:28 PM, Kevin M. White ***@***.***> wrote:﻿
Also looking back further in this thread... your original issue was related to the API call failing... which is not directly related to what super does if it's going to pre-download an update or upgrade.

**Comment by Macjutsu** [comment]
Please try the latest release of `super` as it may resolve this issue: https://github.com/Macjutsu/super/releases/tag/v4.1.0-beta1

**Comment by Macjutsu** [comment]
There have been so many changes to the latest version of `super` that I'm closing this issue.

https://github.com/Macjutsu/super/releases/tag/v5.0.0-beta3

If you find it persists with the latest builds of `super` please open a new issue.
